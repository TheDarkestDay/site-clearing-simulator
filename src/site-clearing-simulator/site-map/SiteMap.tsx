import { Card } from '@material-ui/core';
import React from 'react';
import { useSelector } from 'react-redux';
import { getSiteMap } from '../selectors';
import { SiteCell } from '../site-cell/SiteCell';
import styles from './styles.module.css';

export const SiteMap = () => {
  const siteRows = useSelector(getSiteMap);
  console.log(siteRows);

  return (
    <Card>
      <div className={styles.siteMapWrapper}>
        <div>

        </div>

        <div className={styles.siteMap}>
          {
            siteRows.map((siteCells, i) => {
              return <div key={i} className={styles.siteRow}>
                {
                  siteCells.map((siteCell, j) => <SiteCell key={`${i}-${j}`} type={siteCell} />)
                }
              </div>
            })
          }
        </div>
      </div>
    </Card>
  );
};