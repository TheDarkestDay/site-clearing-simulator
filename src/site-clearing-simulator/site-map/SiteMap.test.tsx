import { shallow } from 'enzyme';
import { SiteMap } from './SiteMap';

describe('SiteMap', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<SiteMap />);

    expect(wrapper).toMatchSnapshot();
  });
});