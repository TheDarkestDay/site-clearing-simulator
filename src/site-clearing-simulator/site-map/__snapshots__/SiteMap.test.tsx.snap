// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteMap should match snapshot 1`] = `ShallowWrapper {}`;
